src/loader.js

```diff
@@ -29,12 +32,17 @@ const findModuleById = (modules, id) => {
 
 export function pitch(request) {
   const query = loaderUtils.getOptions(this) || {};
+
+  validateOptions(schema, query, 'Mini CSS Extract Plugin Loader');
+
   const loaders = this.loaders.slice(this.loaderIndex + 1);
   this.addDependency(this.resourcePath);
   const childFilename = '*'; // eslint-disable-line no-path-concat
   const publicPath =
     typeof query.publicPath === 'string'
       ? query.publicPath
+      : typeof query.publicPath === 'function'
+      ? query.publicPath(this.resourcePath, this.rootContext)
```

Comment:
I think we need handle `/` at end as in https://github.com/webpack-contrib/file-loader/blob/master/src/index.js#L37

Reply:
@evilebottnawi that line in `file-loader` is only on the code path where `publicPath` is a `string`, which is the path we already have; they don't do it in the case that `publicPath` is a `function`. Maybe it's an oversight in `file-loader`? Or perhaps intentional so that the `function` version can have more "power"?

Reply:
hm, maybe you are right, need investigate, anyway handle `/` in string is good idea